src/service/video-manager-impl.js

```diff
@@ -437,6 +556,216 @@ class VideoEntry {
+              this.getPlayingState() == PlayingStates.PAUSED
+              && this.internalElement_.classList.contains(DOCK_CLASS)
+             )
+        )
+      )
+    {
+
+      this.animateDocking_(inViewportHeight);
+
+    } else if (this.minimizePosition_ == MinimizePositions.INVIEW) {
+
+      this.unDockVideo_();
+
+    }
+
+    // Set the initial width and hight when the video is docking
+    // so that we scale relative to the initial video's dimensions
+    if (this.minimizePosition_ != MinimizePositions.INVIEW) {
+      this.internalStyle_.height = this.initialHeight_ + 'px';
+      this.internalStyle_.width = this.initialWidth_ + 'px';
+      this.internalStyle_.maxWidth = this.initialWidth_ + 'px';
+    }
+  }
+
+  /**
+   * Updates the minimization position of the video (in viewport, above or
+   * below viewport) and returns the height of the part of the video that is
+   * currently in the viewport (between 0 and the initial video height).
+   * @param {PositionInViewportEntryDef} newPos
+   * @private
+   * @return {number}
+   */
+  calculateDockableVideoPosition_(newPos) {
+
+    let inViewportHeight = 0;
+
+    if (newPos.positionRect) {
+
+      const docViewTop = newPos.viewportRect.top;
+      const docViewBottom = newPos.viewportRect.bottom;
+
+      const elemTop = newPos.positionRect.top;
+      const elemBottom = newPos.positionRect.bottom;
+
+      // Calculate height currently displayed
+      if (elemTop <= docViewTop) {
+        inViewportHeight = elemBottom - docViewTop;
+        this.minimizePosition_ = MinimizePositions.TOP;
+      } else if (elemBottom >= docViewBottom) {
+        inViewportHeight = docViewBottom - elemTop;
+        this.minimizePosition_ = MinimizePositions.BOTTOM;
+      } else {
+        this.minimizePosition_ = MinimizePositions.INVIEW;
+        inViewportHeight = elemBottom - elemTop;
+      }
+    }
+
+    return inViewportHeight;
+  }
+
+  /**
+   * Performs scroll-bound animations on the video as it is being scrolled
+   * out of the viewport
+   * @param {number} inViewportHeight calculated currently visible video height
+   * @private
+   */
+  animateDocking_(inViewportHeight) {
+    // Minimize the video
+
+    this.video.hideControls();
+    this.internalClasses_.add(DOCK_CLASS);
+    this.internalStyle_.transform = 'scale(' + mapRange(inViewportHeight,
+        0, this.initialHeight_,
+        DOCK_SCALE, 1
+    ) + ')';
+    this.internalStyle_.borderRadius = mapRange(inViewportHeight,
+        this.initialHeight_, 0,
+        0, 6
+    ) + 'px';
+    // Shadow position
+    const sPos = '0px 0px 20px 0px';
+    // Shadow color
+    const sCol = '0, 0, 0'; // rgb black
+    // Shadow opacity
+    const sOpac = mapRange(inViewportHeight,
+        this.initialHeight_, 0,
+        0, 0.3
+    );
+    this.internalStyle_.boxShadow = sPos + ' rgba(' + sCol + ',' + sOpac + ')';
+    this.internalStyle_.left = mapRange(inViewportHeight,
```

Comment:
ditto, try to get the same effect by changing transform-origin rather than `left`